tests/farmer_harvester/test_farmer_harvester.py

```diff
@@ -60,7 +60,7 @@ async def handshake_done() -> bool:
     # Start both services and wait a bit
     await farmer_service.start()
     await harvester_service.start()
-    harvester_service.add_peer(PeerInfo(str(farmer_service.self_hostname), farmer_service._server.get_port()))
+    harvester_service._server.add_peer(PeerInfo(str(farmer_service.self_hostname), farmer_service._server.get_port()))
```

Comment:
accessing the internal member her will make it harder to change implementation details of `harvester_service`, this increases technical debt. I would prefer tests use the public interface.